File: install-page/transform.py
"""Transform coordinates between representations / frames / origins.

Reproducer for the "Transform Coordinates" section of
https://empyrean-dynamics.com/install.
"""

from __future__ import annotations

import empyrean
from empyrean import Frame, Origin


def main() -> None:
    empyrean.initialize()
    orbits = empyrean.query_sbdb(["Apophis"])

    # spielberg:snippet:start
    from empyrean import CartesianCoordinates, KeplerianCoordinates

    # Cometary -> Cartesian (same frame, same origin)
    cart = empyrean.transform_coordinates(orbits.coordinates, CartesianCoordinates)

    # Full transform: representation + frame + origin
    kep = empyrean.transform_coordinates(
        orbits.coordinates,
        KeplerianCoordinates,
        frame=Frame.ICRF,
        origin=Origin.EARTH,
    )
    # spielberg:snippet:end
    _ = (cart, kep)


if __name__ == "__main__":
    main()

File: install-page/transform.rs
//! Transform coordinates between representations / frames / origins.
//!
//! Reproducer for the "Transform Coordinates" section of
//! <https://empyrean-dynamics.com/install>.

use empyrean::{Context, query_sbdb};

fn main() -> empyrean::Result<()> {
    let ctx = Context::from_data_dir(None)?;
    let batch = query_sbdb(&["Apophis"], None)?;
    let orbit = batch.orbits.into_iter().next().expect("not found");

    // empyrean:snippet:start
    use empyrean::{Frame, Origin, Representation};

    // Cometary -> Cartesian (same frame, same origin)
    let cart = ctx.transform_coordinates_single(
        &orbit.state,
        Representation::Cartesian,
        orbit.state.frame,
        orbit.state.origin,
    )?;

    // Full transform: representation + frame + origin
    let kep = ctx.transform_coordinates_single(
        &orbit.state,
        Representation::Keplerian,
        Frame::ICRF,
        Origin::Earth,
    )?;
    // empyrean:snippet:end
    let _ = (cart, kep);
    Ok(())
}
